src/wasm/wasm-type.cpp

```diff
@@ -30,9 +30,9 @@ namespace std {
 template<> class hash<vector<wasm::Type>> {
 public:
   size_t operator()(const vector<wasm::Type>& types) const {
-    auto res = hash<size_t>{}(types.size());
+    uint64_t res = wasm::rehash(0, uint32_t(types.size()));
     for (auto t : types) {
-      wasm::hash_combine(res, t.getID());
+      res = wasm::rehash(res, t.getID());
```

Comment:
I reverted this one to the old implementation before my PR. The old implementation initializes a 32-bit hash value from two 32-bit values, implicitly converts the 32-bit hash value to a 64-bit hash value, and then rehashes with two 64-bit values yielding a 64-bit hash value before returning the resulting 64-bit hash value as `size_t`.

Reply:
Also, this is the only occasion where `t.getID()` is not piped through `std::hash`, with piping through `std::hash` seeming slightly more suboptimal due to conversion from `uint64_t` to `size_t` back to `uin64_t`.